test/jdk/java/awt/Dialog/DialogDisposeLeak.java

```diff
@@ -0,0 +1,199 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.AWTEvent;
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Dialog;
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.FontMetrics;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.Label;
+import java.awt.Toolkit;
+import java.awt.event.MouseEvent;
+import java.awt.event.FocusEvent;
+import java.awt.event.MouseAdapter;
+
+/*
+ * @test
+ * @bug 4193022
+ * @summary Test for bug(s): 4193022, disposing dialog leaks memory
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual DialogDisposeLeak
+ */
+
+public class DialogDisposeLeak {
+    private static final String INSTRUCTIONS = """
+            Click on the Dialog... button in the frame that appears.
+            Now dismiss the dialog by clicking on the label in the dialog.
+
+            Repeat this around 10 times. At some point the label in the frame should change
+            to indicated that the dialog has been garbage collected and the test passed.
+            """;
+
+    public static void main(String args[]) throws Exception {
+        Frame frame = new DisposeFrame();
+        PassFailJFrame.builder()
+                .title("DialogDisposeLeak")
+                .instructions(INSTRUCTIONS)
+                .columns(35)
+                .testUI(frame)
+                .build()
+                .awaitAndCheck();
+    }
+}
+
+class DisposeFrame extends Frame {
+    Label label = new Label("Test not passed yet");
+
+    DisposeFrame() {
+        super("DisposeLeak test");
+        setLayout(new FlowLayout());
+        Button btn = new Button("Dialog...");
+        add(btn);
+        btn.addActionListener(ev -> {
+                    Dialog dlg = new DisposeDialog(DisposeFrame.this);
+                    dlg.setVisible(true);
+                }
+        );
+        add(label);
+        pack();
+    }
+
+    public void testOK() {
+        label.setText("Test has passed. Dialog finalized.");
+    }
+}
+
+class DisposeDialog extends Dialog {
+    DisposeDialog(Frame frame) {
+        super(frame, "DisposeDialog", true);
```

Comment:
The dialog to click on is too far away from the parent frame, it will be easier to click 10 times if it is just under the mouse cursor.

```suggestion
        super(frame, "DisposeDialog", true);
        setLocation(frame.getX(), frame.getY());
```